node_package/src/clientStartup.ts

```diff
@@ -135,13 +135,31 @@ function domNodeIdForEl(el: Element): string {
  * delegates to a renderer registered by the user.
  * @param el
  */
+
+type ComponentCache = { [key: string]: boolean };
+let components: { [key: string]: ComponentCache; } = {};
+
 function render(el: Element, railsContext: RailsContext): void {
   const context = findContext();
   // This must match lib/react_on_rails/helper.rb
   const name = el.getAttribute('data-component-name') || "";
   const domNodeId = domNodeIdForEl(el);
   const props = (el.textContent !== null) ? JSON.parse(el.textContent) : {};
   const trace = el.getAttribute('data-trace') === "true";
+  const path = window.location.pathname;
+  const locationMounted = components[path];
+
+  if (locationMounted) {
+    const componentMounted = locationMounted[props.componentName];
+    if (componentMounted) return;
```

Comment:
returning seems to work great for us. I'm not positive if we do in fact need to continue execution of some of the code in the remainder of this function that already existed. But we can't run all of it.